pandas/tests/extension/test_datetime.py

```diff
@@ -162,6 +162,20 @@ def _compare_other(self, s, data, op_name, other):
         # with (some) integers, depending on the value.
         pass
 
+    def test_compare_with_Categorical(self):
```

Comment:
can you parameterize this over the  types (date_range and such),

also pls add a case for !=